src/pixel/url-collector.ts

```diff
@@ -29,6 +29,17 @@ export function collectUrl(state: UrlState): [string, boolean, string[]] {
   return [url.toString(), pathRemoved, blockedParams]
 }
 
+export function stripQueryAndPath(pageUrl?: string): string {
```

Comment:
Empty string will be stripped by the query builder default. In case this is the desired behaviour, I propose more explicitly communicating that in the types by returning undefined

Reply:
Yes, desired behaviour. Changed the signature to `stripQueryAndPath(pageUrl: string): string` and the impl uses `onNonNull` now.